From dd4dd5ec55610a8d2c94667dc7a19ab742091528 Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Fri, 30 Oct 2020 22:54:37 +0200
Subject: [PATCH 190/269] Pass U-boot's DTB address to OP-TEE.

If given the address of a DTB, OP-TEE can inject platform configuration
nodes during its initialization. These nodes will be passed down to the
Linux kernel during U-boot's initialization to ensure communication
between the Linux Kernel and OP-TEE afterwards.

Issue: ALB-5525
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32g/s32g_bl31.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plat/nxp/s32g/s32g_bl31.c b/plat/nxp/s32g/s32g_bl31.c
index 7d2434782..f98f414ac 100644
--- a/plat/nxp/s32g/s32g_bl31.c
+++ b/plat/nxp/s32g/s32g_bl31.c
@@ -180,6 +180,7 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
 	bl32_image_ep_info.pc = S32G_BL32_BASE;
 	bl32_image_ep_info.spsr = 0;
 	bl32_image_ep_info.args.arg0 = MODE_RW_64;
+	bl32_image_ep_info.args.arg3 = BL33_DTB;
 #endif
 }
 
-- 
2.17.1

